From bug http://bugs.freedesktop.org/show_bug.cgi?id=12434

"It's a perfectly fine patch to put in distributions for the time being"
  -- Daniel Stone, Comment #18

--- xorg-server-1.4/dix/devices.c.orig  2007-10-19 17:04:51.000000000 -0200
+++ xorg-server-1.4/dix/devices.c       2007-10-23 17:50:38.000000000 -0200
@@ -294,7 +294,17 @@ CoreKeyboardBell(int volume, DeviceIntPt
 static void
 CoreKeyboardCtl(DeviceIntPtr pDev, KeybdCtrl *ctrl)
 {
+#ifdef XKB
+    if (!noXkbExtension) {
+       DeviceIntPtr dev = (DeviceIntPtr)
+           pDev->devPrivates[CoreDevicePrivatesIndex].ptr;
+
+       if (dev && dev->key && dev->key->xkbInfo && dev->key->xkbInfo->kbdProc)
+           (*dev->key->xkbInfo->kbdProc)(dev, ctrl);
+    }
+#else
     return;
+#endif
 }

 /**
